torch/distributions/relaxed_categorical.py

```diff
@@ -0,0 +1,110 @@
+    return log_softmax(x, *args, **kwargs)
+
+
+class ExpRelaxedOneHotCategorical(Distribution):
+    r"""
+    Creates a relaxed one-hot categorical distribution parameterized by `probs` and `temperature`.
+    Returns a log of a point in the simplex. Based on the interface to OneHotCategorical.
+
+    Implementation based on [1].
+
+    See also: :func:`torch.distributions.OneHotCategorical`
+
+    Example::
+
+        >>> m = OneHotCategorical(torch.Tensor([ 0.25, 0.25, 0.25, 0.25 ]))
+        >>> m.sample()  # equal probability of 0, 1, 2, 3
+         0
+         0
+         1
+         0
+        [torch.FloatTensor of size 4]
+
+    Args:
+        probs (Tensor or Variable): event probabilities
+
+    [1] The Concrete Distribution: A Continuous Relaxation of Discrete Random Variables
+    (Maddison et al, 2017)
+
+    [2] Categorical Reparametrization with Gumbel-Softmax
+    (Jang et al, 2017)
+    """
+    params = {'probs': constraints.simplex}
+    support = constraints.real
+    has_rsample = True
+
+    def __init__(self, temperature, probs=None, logits=None):
+        self._categorical = Categorical(probs, logits)
+        self.temperature = temperature
+        batch_shape = self._categorical.batch_shape
+        event_shape = self._categorical.param_shape[-1:]
+        super(ExpRelaxedOneHotCategorical, self).__init__(batch_shape, event_shape)
+
+    def _new(self, *args, **kwargs):
+        return self._categorical._new(*args, **kwargs)
+
+    @property
+    def param_shape(self):
+        return self._categorical.param_shape
+
+    @property
+    def logits(self):
+        return self._categorical.logits
+
+    @property
+    def probs(self):
+        return self._categorical.probs
+
+    def rsample(self, sample_shape=torch.Size()):
+        sample_shape = torch.Size(sample_shape)
+        uniforms = clamp_probs(self.logits.new(self._extended_shape(sample_shape)).uniform_())
+        gumbels = uniforms.log().mul(-1).log().mul(-1)
+        return _log_softmax((self.logits + gumbels) / self.temperature, dim=-1)
+
+    def log_prob(self, value):
+        K = self._categorical._num_events
+        self._validate_log_prob_arg(value)
+        logits, value = broadcast_all(self.logits, value)
+        log_scale = (self.logits.new(1).fill_(K).lgamma() -
+                     self.temperature.log().mul(-(K - 1)))
+        score = _log_softmax(logits - value.mul(self.temperature), dim=-1).sum(-1)
+        return (score + log_scale)
+
+
+class RelaxedOneHotCategorical(TransformedDistribution):
+    r"""
+    Documentation goes here
+    """
+    params = {'probs': constraints.simplex}
+    support = constraints.simplex
+    has_rsample = True
+
+    def __init__(self, temperature, probs=None, logits=None):
+        super(RelaxedOneHotCategorical, self).__init__(ExpRelaxedOneHotCategorical(temperature, probs, logits),
+                                                       ExpTransform())
```

Comment:
I would expect either the transform to be `ExpTransform().inv` or the base distribution to be `LogRelaxedOneHotCategorical`. Is the latter a reasonable renaming, analogous to `LogNormal`?

Reply:
Right, I agree with `LogRelaxedOneHotCategorical` as well. However, TF uses the version I have because that's what the paper names it. My thinking is that it's probably more important just to document it and be consistent with the other interfaces, since no user will need to see it.

Reply:
Sounds good, let's keep it standard.